Tests/ParseSwiftTests/ParseUserTests.swift

```diff
@@ -1083,59 +1083,75 @@ class ParseUserTests: XCTestCase { // swiftlint:disable:this type_body_length
-                let encoded = try ParseCoding.jsonEncoder().encode(logoutResponse)
-                return MockURLResponse(data: encoded, statusCode: 200, delay: 0.0)
+                try User.logout()
+                if let userFromKeychain = BaseParseUser.current {
+                    XCTFail("\(userFromKeychain) wasn't deleted from Keychain during logout")
+                }
+                DispatchQueue.main.async {
+                    if let installationFromKeychain = BaseParseInstallation.current {
+                        if installationFromKeychain.installationId == oldInstallationId {
+                            XCTFail("\(installationFromKeychain) wasn't deleted then created in Keychain during logout")
+                        }
+                    }
+                }
             } catch {
-                return nil
-            }
-        }
-        do {
-            try User.logout()
-            if let userFromKeychain = BaseParseUser.current {
-                XCTFail("\(userFromKeychain) wasn't deleted from Keychain during logout")
-            }
-
-            if let installationFromKeychain = BaseParseInstallation.current {
-                XCTFail("\(installationFromKeychain) wasn't deleted from Keychain during logout")
+                XCTFail(error.localizedDescription)
             }
-        } catch {
-            XCTFail(error.localizedDescription)
         }
     }
 
     func logoutAsync(callbackQueue: DispatchQueue) {
 
         let expectation1 = XCTestExpectation(description: "Logout user1")
-        User.logout(callbackQueue: callbackQueue) { result in
 
-            switch result {
+        DispatchQueue.main.async {
+            let oldInstallationId = BaseParseInstallation.current?.installationId
+            User.logout(callbackQueue: callbackQueue) { result in
 
-            case .success:
-                if let userFromKeychain = BaseParseUser.current {
-                    XCTFail("\(userFromKeychain) wasn't deleted from Keychain during logout")
-                }
+                switch result {
 
-                if let installationFromMemory: CurrentInstallationContainer<BaseParseInstallation>
-                    = try? ParseStorage.shared.get(valueFor: ParseStorage.Keys.currentInstallation) {
-                    XCTFail("\(installationFromMemory) wasn't deleted from memory during logout")
-                }
+                case .success:
+                    if let userFromKeychain = BaseParseUser.current {
+                        XCTFail("\(userFromKeychain) wasn't deleted from Keychain during logout")
+                    }
+                    DispatchQueue.main.async {
+                        if let installationFromMemory: CurrentInstallationContainer<BaseParseInstallation>
+                            = try? ParseStorage.shared.get(valueFor: ParseStorage.Keys.currentInstallation) {
+                            if installationFromMemory.installationId == oldInstallationId {
```

Comment:
What do you think about extending the test case here and also checking if `installationFromKeychain.currentInstallation != nil`?